### PR TITLE
Add a flag to allow additional object properties

### LINKED ISSUE #9: setting additionalProperties to true
Not sure if this belongs in the vega fork or the original repo.
Looking at the code, it's not clear that 'additionalProperties' can ever be set to true in the generated schema. 

Best I could figure was this function https://github.com/vega/typescript-to-json-schema/blob/master/src/NodeParser/InterfaceNodeParser.ts#L65 and it'll always return false. 

There seem to be no test cases to the counter. 

Would appreciate clarification and/or some idea of how to go about fixing this. Happy to submit a PR once given a scent to follow.

### PR DESCRIPTION
**Pull request by tamj0rd2**
fixes #9

### CODE REVIEW

**Comment by tamj0rd2** [comment]
`yarn build` and `yarn test` both passed

File: README.md [hunk 1]
```diff
@@ -104,6 +104,10 @@ Run the schema generator via `node main.js`.
 
 -o, --out
     Specify the output file path. Without this option, the generator logs the response in the console.
+
+--allow-additional-properties  <true|false>
+    true: Allow objects to have additional properties
+    false (default): Do not allow objects to have additional properties
```
Comment (domoritz) [inline README.md:110]: That's not entirely true. We still allow additional properties but you have to write something like

```ts
{
  [key: string]: number
}
```

Can you rewrite the description to be more precise?

File: README.md [hunk 2]
```diff
@@ -104,6 +104,10 @@ Run the schema generator via `node main.js`.
 
 -o, --out
     Specify the output file path. Without this option, the generator logs the response in the console.
+
+--allow-additional-properties  <true|false>
```
Comment (domoritz) [inline README.md:108]: Can we find a different name here? `allow-additional-properties: false` to me sounds like you cannot use additional properties at all. Maybe `additional-properties-by-default` or just `additional-properties` (with a precise description) are better. What do you think?

Comment (tamj0rd2) [reply README.md:108]: `additional-properties` sounds good to me

File: README.md (hunk 1 above)
Comment (tamj0rd2) [reply README.md:110]: How about this?

```
--additional-properties  <true|false>
    Controls whether or not to allow additional properties for objects that have no index signature.

    true: Additional properties are allow
    false (default): Additional properties are not allowed
```

Comment (domoritz) [reply README.md:110]: Love it. Thanks!

Additional properties are allow -> Additional properties are allowed

File: README.md (hunk 2 above)
Comment (domoritz) [reply README.md:108]: Let's also rename the variable used in the code.

Comment (tamj0rd2) [reply README.md]: All done, test and build are both passing

**Comment by domoritz** [comment]
Thank you!

**Comment by yliu-d** [comment]
We upgraded to 0.70.0 and `additionalProperties: false` disappeared from the generated JSON schema, when we didn't set this flag.

**Comment by domoritz** [comment]
Ahh, sorry @yliu-d, I didn't check the code carefully enough. Fixed in 0.70.1.

**Pull request merged**